fenix/app/src/main/java/org/mozilla/fenix/browser/TranslationsIconBinding.kt

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.browser
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.selector.findTab
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.TranslationsState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.translate.Language
+import mozilla.components.concept.engine.translate.initialFromLanguage
+import mozilla.components.concept.engine.translate.initialToLanguage
+import mozilla.components.lib.state.helpers.AbstractBinding
+
+/**
+ * Helper for observing Translation state from browserStore.
+ * @param browserStore Containing the information about [TranslationsState].
+ * @param sessionId Current open tab session id.
+ * @param onIconChanged Invoked when icon visibility, "from" language selected,
+ * "to" language selected, and the translated flag are changing.
+ */
+class TranslationsIconBinding(
+    private val browserStore: BrowserStore,
+    private val sessionId: String,
+    private val onIconChanged: (
```

Comment:
```suggestion
    private val onStateUpdated: (
```
Just thinking about the naming of this a bit more - while the icon will be changed depending on how the consumer uses this lambda, this binding class is actually pretty generic and only serving as a callback passing in the new translation state information to the consumer. So, I think we should actually remove any mention of `icon` and treat this as reusable for anything that cares about reacting to these state changes.

Reply:
Done